cpp/OpenSqliteHo.cpp

```diff
@@ -0,0 +1,94 @@
+//
+// Created by jplc on 4/2/24.
+//
+
+#include "OpenSqliteHo.h"
+#include "bridge.h"
+#include "macros.h"
+#include "types.h"
+
+namespace jsi = facebook::jsi;
+using std::string;
+
+namespace opsqlite {
+
+jsi::Function OpenSqliteHo::open(jsi::Runtime &rt,
+                                 const std::string &basePath) {
+  return HOSTFN("open", 3) {
```

Comment:
In case you still need some help: You are kinda on the right path, but this is not quite correct. You need the `get`, `set` and `getPropertyNames` function for the HostObject to work. If you want to declare further internal functions, there is no need to return a HOSTFN. You can declare the HOSTFNs on the header file as `std::function` though, that would be a cool optimization actually.